index.js

```diff
@@ -6,6 +6,7 @@ function HashBase (blockSize) {
   Transform.call(this)
 
   this._block = new Buffer(blockSize)
+  this._block.fill(0)
```

Comment:
ok so this one is unnecessary since this wouldn't be private information we needed to hide.

hashes can come in 3 states

- not used
- in progress
- completed 

currently there is no way to distinguish the 3 by looking at the buffer, this pull would change it so that completed and not used were zero filled while in progress is not on the grounds that 

 > internal state before and after using should be same

I have no reason to believe that and while I'd be happy to be convinced otherwise it just seems like cargo culting some pattern or something